### PR TITLE
Fix next schedule time duration

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind bug
/kind regression
/sig apps

#### What this PR does / why we need it:
Initially `mostRecentScheduleTime` method was returning a number of missed schedules, but that turned out to be not reliable for some complex schedules. For example, those which are being run only during week days. The second approach was to only return a boolean indicating the too many missed information. It turns out that we need to return all three values: none missed, few missed and many missed, to let consumers know what to do, but don't leak the wrong number out of mostRecentScheduleTime`.

This is alternative approach to https://github.com/kubernetes/kubernetes/pull/120738

#### Special notes for your reviewer:
/assign @sushanth0910

#### Does this PR introduce a user-facing change?
```release-note
Fixes a 1.25+ regression calculating the requeue time in the cronjob controller, which results in properly handling failed/stuck jobs
```

### CODE REVIEW

File: pkg/controller/cronjob/utils_test.go [hunk 1]
```diff
@@ -611,6 +616,83 @@ func TestMostRecentScheduleTime(t *testing.T) {
...
+				},
+				Spec: batchv1.CronJobSpec{
+					Schedule: "0 12 * * 4",
+				},
+				Status: batchv1.CronJobStatus{
+					LastScheduleTime: &metav1TwoHoursLater,
+				},
+			},
+			now:              *deltaTimeAfterTopOfTheHour(19*24*time.Hour + 1*time.Hour + 30*time.Minute),
+			expectedDuration: 48*time.Hour + 30*time.Minute + nextScheduleDelta,
```
Comment (atiratree) [inline pkg/controller/cronjob/utils_test.go:676]: can you please add an additional simple test case without `.status.lastScheduleTime` that has not run yet? So we hit this line as well:

```
if missedSchedules == noneMissed {
```

Comment (soltysh) [reply pkg/controller/cronjob/utils_test.go:676]: good idea - updated

**Pull request merged**